tensorflow_addons/optimizers/conditional_gradient_test.py

```diff
@@ -67,109 +67,141 @@ def test_like_dist_belief_nuclear_cg01():
+    # for 'GPU' devices. However, tf.half is not registered in this case,
+    # based on issue #347.
+    # Thus, we will call the "_dtypes_to_test" function.
+    #
+    # TODO:
+    #       Wait for the solving of issue #347. After that, we will test
+    #       for the dtype to be tf.half, with 'GPU' devices.
+    for dtype in _dtypes_to_test(use_gpu=tf.test.is_gpu_available()):
+        var0 = tf.Variable([[1.0, 2.0]], dtype=dtype)
 
-            if not use_callable_params:
-                learning_rate = learning_rate()
-                lambda_ = lambda_()
-
-            cg_opt = cg_lib.ConditionalGradient(
-                learning_rate=learning_rate, lambda_=lambda_, ord=ord
-            )
-            cg_update = cg_opt.apply_gradients(zip([grads0, grads1], [var0, var1]))
-
-            if not tf.executing_eagerly():
-                self.evaluate(tf.compat.v1.global_variables_initializer())
-                # Fetch params to validate initial values
-                self.assertAllClose([1.0, 2.0], self.evaluate(var0))
-                self.assertAllClose([3.0, 4.0], self.evaluate(var1))
+        def loss():
+            x = tf.constant([[4.0], [5.0]], dtype=dtype)
+            pred = tf.matmul(tf.nn.embedding_lookup([var0], [0]), x)
+            return pred * pred
+
+        # the gradient based on the current loss function
+        grads0_0 = 32 * 1.0 + 40 * 2.0
+        grads0_1 = 40 * 1.0 + 50 * 2.0
+        grads0 = tf.constant([[grads0_0, grads0_1]], dtype=dtype)
+        norm0 = tf.math.reduce_sum(grads0 ** 2) ** 0.5
 
-            # Check we have slots
-            self.assertEqual(["conditional_gradient"], cg_opt.get_slot_names())
-            slot0 = cg_opt.get_slot(var0, "conditional_gradient")
-            self.assertEquals(slot0.get_shape(), var0.get_shape())
-            slot1 = cg_opt.get_slot(var1, "conditional_gradient")
-            self.assertEquals(slot1.get_shape(), var1.get_shape())
+        learning_rate = 0.1
+        lambda_ = 0.1
+        ord = "fro"
+        opt = cg_lib.ConditionalGradient(
+            learning_rate=learning_rate, lambda_=lambda_, ord=ord
+        )
+        _ = opt.minimize(loss, var_list=[var0])
+        test_utils.assert_allclose_according_to_type(
+            [
+                [
+                    1.0 * learning_rate
+                    - (1 - learning_rate) * lambda_ * grads0_0 / norm0,
+                    2.0 * learning_rate
+                    - (1 - learning_rate) * lambda_ * grads0_1 / norm0,
+                ]
+            ],
+            var0.numpy(),
+        )
 
-            if not tf.executing_eagerly():
-                self.assertFalse(slot0 in tf.compat.v1.trainable_variables())
-                self.assertFalse(slot1 in tf.compat.v1.trainable_variables())
 
-            if not tf.executing_eagerly():
-                self.evaluate(cg_update)
+def do_test_basic_frobenius(dtype, use_resource, use_callable_params=False):
```

Comment:
You can remove the `use_callable_params` because it's always `True`.

Reply:
That makes sense. Thanks @gabrieldemarmiesse.